addons/point_of_sale/static/src/js/db.js

```diff
@@ -214,6 +214,14 @@ var PosDB = core.Class.extend({
             if(product.barcode){
                 this.product_by_barcode[product.barcode] = product;
             }
+            self = this;
```

Comment:
it is probably not what must be done, because a packaging can have a quantity. So you should modify the scanning functions to check values in your variable product_packagings, and then add the correct qty of product on the pos order.